Providers/nxOMSAutomationWorker/automationworker/3.x/worker/runbook.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+# ====================================
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+# ====================================
+
+"""Runbook module."""
+
+import os
+
+import configuration3 as configuration
+import gpg
+import iohelper
+import jrdsclient
+import linuxutil
+from workerexception import *
+
+definition_kind_int_to_str = {0: "Unknown",
+                              1: "Script",
+                              2: "Workflow",
+                              3: "Graph",
+                              4: "Configuration",
+                              5: "PowerShell",
+                              6: "PowerShellWorkflow",
+                              7: "GraphPowerShell",
+                              8: "GraphPowerShellWorkflow",
+                              9: "Python2",
+                              10: "Python3",
+                              11: "Bash"}
+
+
+class Runbook(object):
+    def __init__(self, runbook_data):
+        """:type runbook_data: jrdsclient.RunbookData"""
+        self.runbook_data = runbook_data
+        self.definition_kind = runbook_data.runbook_definition_kind
+        self.definition_kind_str = definition_kind_int_to_str[runbook_data.runbook_definition_kind]
+        self.runbook_file_path = ""
+
+        # should be overwritten by derived class
+        self.file_extension = ""
+
+    def initialize(self):
+        """Initializes the runbook."""
+        self.write_to_disk()
+        if configuration.get_enforce_runbook_signature_validation():
+            self.validate_signature()
+
+    def write_to_disk(self):
+        """Writes the runbook's definition to disk."""
+        file_name = self.runbook_data.name + self.runbook_data.runbook_version_id + self.file_extension
+        self.runbook_file_path = os.path.join(configuration.get_working_directory_path(), file_name)
+        # In python2 strings and bytes are treated the same. Hence the encoding had no affect on the string to be written. But in python3 encoding changes the string to bytes.
+        # So there is no need because it is already encoded.
+        runbook_definition = str(self.runbook_data.definition)
+        if linuxutil.is_posix_host() is True:
+            # replace dos end of line to unix end of line
+            runbook_definition = runbook_definition.replace("\r\n", "\n")
+        iohelper.write_to_file(self.runbook_file_path, data=runbook_definition, mode="w")
+
+    def validate_signature(self):
```

Comment:
Let's validate the signature validation. We can test using update management, automation assets and an unsigned runbook.

Reply:
Tested the automation assets, unsigned runbooks execution and UM as well.